src/deconstruction.ts

```diff
@@ -0,0 +1,722 @@
+import * as cdk from 'aws-cdk-lib';
+import { CfnResource, Tags } from 'aws-cdk-lib';
+import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import { IConstruct } from 'constructs';
+import * as reflect from 'jsii-reflect';
+import {
+  isConstruct,
+  isDataType,
+  isEnumLikeClass,
+  isSerializableInterface,
+  SchemaContext,
+  schemaForPolymorphic,
+} from './jsii2schema';
+
+export interface Tag {
+  Key: string;
+  Value: string;
+}
+
+export interface Override {
+  ResourcePath: string;
+  Update: { Path: string; Value: any };
+  Delete?: { Path: string };
+  RemoveResource?: boolean;
+}
+
+export interface DeclarativeResource {
+  Type: string;
+  Properties?: any;
+  Overrides?: Override[];
+  DependsOn?: string | string[];
+  Tags?: Tag[];
+}
+
+export function resolveType(fqn: string) {
+  const [mod, ...className] = fqn.split('.');
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  const module = require(mod);
+
+  let curr = module;
+  while (true) {
+    const next = className.shift();
+    if (!next) {
+      break;
+    }
+    curr = curr[next];
+    if (!curr) {
+      throw new Error(`unable to resolve class ${className}`);
+    }
+  }
+
+  return curr;
+}
+
+export interface ParsedIntrinsic {
+  readonly name: string;
+  readonly value: any;
```

Comment:
`unknown` ? If you don't know, you don't know, and `unknown` will make TypeScript force you to do type checks before accessing the value.